hathor/manager.py

```diff
@@ -149,6 +149,10 @@ def __init__(self, reactor: Reactor, peer_id: Optional[PeerId] = None, network:
             self.log.debug('enable wallet indexes')
             self.tx_storage.indexes.enable_address_index(self.pubsub)
             self.tx_storage.indexes.enable_tokens_index()
+        if utxo_index and self.tx_storage.with_index:
```

Comment:
I feel we should add a method here `enable_utxo_index()`, instead of adding one more parameter to the constructor. I'm good to move forward like this since we need to refactor the whole initialization structure and improve the dependencies injection.

Reply:
I don't really like the `enable_*` structure, I really think we shouldn't move in that direction. Mostly because we can't just `enable_*` and `disable_*` freely, and running `enable_*` _after_ initialization will cause errors that are hard to track, to properly support that we'd need more structures (like storing which indexes have already been initialized, and if an index is enabled after initialization it would have to either be rejected with an error or properly reinitialized).

Particularly, if we had an `HathorManager.enable_utxo_index` method, it would be really tempting to use it on tests that already have a manager instantiated, but this will cause problems, because it's very common for blocks and transactions to be created immediately after instantiating a manager, and the `enable_utxo_index` (without careful protections) will miss those transctions, and even if none was added, it would have missed the genesis.

We sort of have this "problem" on the TransactionStorage, however `enable_*` is _exclusively_ used in very few situations, but not without causing problems, the index resetting mechanism had to be changed to account for indexes that were enabled and the information on which indexes were enabled was "lost" or not accessible without breaking encapsulation patterns.

I think that a better structure is having constructor classes (similar to factory, but only for one instantiation, instead of multiple, which simplifies some things).

While we don't have a better structure, I prefer to use a pattern that is the least disruptive as possible, instead of adding yet another pattern half-implemented. So, for now, as unfortunate as it is, I really prefer adding one more parameter to `HathorManager.__init__`